data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -323,6 +327,10 @@ private AccumulatingBulkRequest<BulkOperationWrapper, BulkRequest> createBulkReq
                         LOG.debug("Received version conflict from OpenSearch: {}", bulkItemResponse.error().reason());
                         bulkOperation.releaseEventHandle(true);
                     } else {
+                        if (RestStatus.FAILED_DEPENDENCY.getStatus() == bulkItemResponse.status()) {
```

Comment:
How hard would it be to test these conditions? If not too hard, let's verify the behavior.